### DISCUSSION TITLE
Long running process (bulk insertion) blocks the other pages

### DISCUSSION THREAD

**Discussion by deanilvincent** [symfony/symfony#54537] [Q&A] [answered]
Hi, 
How can we prevent the UI from blocking when running a long process of database insertion?
We have a process that needs a bulk insertion. The insertion logic inserts hundreds or even thousands of records. We found a package from Symfony which the symfony process component 
[The component process](https://symfony.com/doc/current/components/process.html)

I setup it up like this using the `process->start()` but the insertion of records doesn't work.
```php
 $cmd = 'php ' . $this->getParameter('PROJECT_DIR') . '/bin/console long-running-command';
 $process = Process::fromShellCommandline($cmd);
 $process->setTimeout(3600);
 $process->start();
```
I tried to use the `$process->run()` and it started to insert records but since this is synchronous, it blocks the other UI. Basically, while the long running process is running, when I navigate to the other pages, the web browser is loading. On .NET, there's a built-in async function. On PHP, I found the [ampphp](https://github.com/Amphp/Amp). 

My question is, how can we properly handle or configure a long running process on Symfony without blocking other UIs? The idea is, I want to trigger the bulk insertion and run it on the background without affecting the loading of the pages etc.

Thank you in advance!

**Comment by stof** [comment, marked as answer]
Use the messenger component to run this asynchronously in a separate worker process.

Your code starting a CLI process from the web process won't solve the issue because a child process does not outlive its parent process, which means it is keeping the web process alive until the end of the long-running command.

**Reply by ghost** [reply]
I agree with this answer.
